### ISSUE TITLE
Invalid syntax in function and class definition does not report invalid syntax but raise AttributeError

### ISSUE THREAD

**Issue by fnielsen** [PyCQA/pydocstyle#159]
If there is invalid syntax pep257 may report "invalid syntax in file test.py". 

However, with syntax error in the function of class definition an AttributeError is raised: 

```
$ pep257 test.py
Traceback (most recent call last):
  File "/usr/local/bin/pep257", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/dist-packages/pep257.py", line 1605, in main
    sys.exit(run_pep257())
  File "/usr/local/lib/python2.7/dist-packages/pep257.py", line 1275, in run_pep257
    errors.extend(check((filename,), select=checked_codes))
  File "/usr/local/lib/python2.7/dist-packages/pep257.py", line 1216, in check
    for error in PEP257Checker().check_source(source, filename):
  File "/usr/local/lib/python2.7/dist-packages/pep257.py", line 1313, in check_source
    module = parse(StringIO(source), filename)
  File "/usr/local/lib/python2.7/dist-packages/pep257.py", line 259, in __call__
    return self.parse_module()
  File "/usr/local/lib/python2.7/dist-packages/pep257.py", line 409, in parse_module
    children = list(self.parse_definitions(Module, all=True))
  File "/usr/local/lib/python2.7/dist-packages/pep257.py", line 348, in parse_definitions
    yield self.parse_definition(class_._nest(self.current.value))
  File "/usr/local/lib/python2.7/dist-packages/pep257.py", line 445, in parse_definition
    if self.current.kind in (tk.NEWLINE, tk.COMMENT):
AttributeError: 'NoneType' object has no attribute 'kind'
```

`test.py` could be: 

```
def function()
    pass
```

or 

```
class Class()
    pass
```

**Comment by sigmavirus24** [comment]
So:

1. pep257 is just using the tokenize module from the standard library which doesn't validate syntax (which is why this happens)
2. pep257 would need to compile the code (using the ast module most likely) to catch this for you
3. Other tools (e.g., pyflakes, pylint, flake8) can and will catch this

pep257 should probably attempt to prevent this but shouldn't explicitly check for invalid syntax. It should just quit when it can no longer tokenize or parse something.

**Comment by fnielsen** [comment]
I have attempted a fix: https://github.com/GreenSteam/pep257/compare/master...fnielsen:issue159 I am not sure whether it works for all cases. To me it seems that `leapfrog` should raise if it does not reach its required token?

**Comment by aphedges** [comment]
When running on both of these files, I now get the error `WARNING: Error in file test.py: Cannot parse file.` A check to ensure proper syntax has been implemented since 3041dcbd64ceb2d80fab9f3caebbcc5cb777a2a5, so this issue is no longer a problem.
